keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -263,17 +482,166 @@ When downgrading from a release with this feature, to a release without maxUnava
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+#### Alpha
+
+- Initial support for maxUnavailable in StatefulSets added. Disabled by default with default value of 1.
+
+#### Beta
+
+- Enabled by default with default value of 1 with upgrade downgrade tested at least manually.
 
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+#### Upgrade
+
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+  - Nothing special, it's backward compatibility
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+  - This feature is enabled by default in beta so you can configure the maxUnavailable to meet your requirements
+
+#### Downgrade
+
+when downgrade from a release `maxUnavailable` enabled to a release it's disabled:
+
+- If we downgrade to a release even doesn't have this field, then there's no way to discover it.
+- Else, if we create statefulset with maxUnavailable configured, it will be dropped to nil automatically.
```

Comment:
That's not what we're doing.
The convention we're using is:
- if you try to set this field, it's silently dropped
- if the object with this field set already exists, we maintain it as it is - it's just controller that should ignore it

Reply:
+1